tests/TestUtils.ts

```diff
@@ -0,0 +1,23 @@
+export class TestUtils {
+  public static comparator(sortKey: string): (object1: object, object2: object) => number {
+    return (object1: object, object2: object) => {
+      if (object1[sortKey] < object2[sortKey]) return -1;
+      if (object1[sortKey] > object2[sortKey]) return 1;
+      return 0;
+    };
+  }
+
+  public static toMatchArrayContents(received: any[], expected: any[], sortKey = 'uuid') {
```

Comment:
https://jestjs.io/docs/en/expect#expectarraycontainingarray

Reply:
I think this is slightly different - we need a way to have `expectArrayContaining` but with objects that partially match (`expectArrayContaining` expects exact matches).  In the case with feedback, we're comparing two arrays of `Feedback[]` and `PublicFeedback[]`, and in other tests, we'd compare the `*Model[]` array to `Public*[]` which also only partially match. There doesn't seem to be a way to do this through jest's current API

Reply:
Why not map the Model arrays to the PublicModel arrays?

Reply:
oh yeah we can do that for the other tests, but the feedback factory creates `Feedback` objects rather than `FeedbackModel` ones because `FeedbackModel` relies on a `UserModel` to submit it. 

I personally think it better mimics the actual flow of feedback to have `Feedback` objects created in the factory and `PortalState::submitFeedback(feedback: Feedback, user: UserModel)` actually attaching a user to it, but I suppose having  `FeedbackFactory::create(n: number, user: UserModel): FeedbackModel[]` instead doesn't hurt too

Reply:
Current FeedbackFactory's fine; I meant for the assertions, we can map Model[] to PublicModel[] instead of creating our own matcher. I think the tests should use the controller to submit feedback instead though.

Reply:
I was thinking about having a different test for each controller alone so that only one thing's tested at a time, so all data that would need to be inserted for a test is done with `PortalState` and just 1 controller is tested rather than 2+.

Reply:
> I was thinking about having a different test for each controller alone so that only one thing's tested at a time

Do you mean one test per route/controller method? Most of the POST/GET routes are pretty basic CRUD operations so I think those are fine to test together, e.g. submit a feedback and then get it. Looking at `FeedbackService` the only code that's executed in that flow is pretty simple stuff like creating and inserted via TypeORM and then a basic `SELECT` query.